src/main/java/com/github/manolo8/darkbot/Bot.java

```diff
@@ -32,10 +41,86 @@ public static void main(String[] args) throws IOException {
         LibSetup.setupLibraries();
         StartupParams params = new StartupParams(args);
 
+        checkUniqueInstance(params); //java 9
         checkJavaVersion(params);
         SwingUtilities.invokeLater(() -> new Main(params));
     }
 
+    private static void checkUniqueInstance(StartupParams params) throws IOException {
+        if (params.has(StartupParams.LaunchArg.NO_WARN)) return;
+
+        Path filePath = Paths.get("curr.pid");
+        long currentPid = ProcessHandle.current().pid();
+        ProcessHandle processHandle = ProcessHandle.current();
+        ProcessHandle.Info processInfo = processHandle.info();
+        long currentStartTime = processInfo.startInstant().map(Instant::toEpochMilli).orElse(0L);
+
+        if (Files.exists(filePath)) {
+            List<String> fileContent = Files.readAllLines(filePath, StandardCharsets.UTF_8);
+
+            if (!fileContent.isEmpty()) {
```

Comment:
there is no need for this check, it will simply perform a loop with 0 iterations without it, which reduces identation as a whole